libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/backpress/DefaultOnBackPressedCallback.kt

```diff
@@ -13,70 +15,68 @@ import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
 import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
 import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
 import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
-import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
 /**
- * Key listener for the [NavigationView].
+ * Back pressed callback for the [NavigationView].
  *
  * On back pressed will update the navigation state.
  *
  * (FreeDrive) <- (DestinationPreview) <- (RoutePreview) <- (ActiveNavigation)
  *             <- (Arrival)
  */
-@ExperimentalPreviewMapboxNavigationAPI
-internal class OnKeyListenerComponent(
+internal class DefaultOnBackPressedCallback private constructor(
     private val store: Store,
-    private val view: View,
-    private val delegateOnKeyListener: View.OnKeyListener? = null
-) : UIComponent() {
```

Comment:
Even though this does not access MapboxNavigation at the moment, I still think it should be a `UIComponent`

Specifically because we're building tools like `attachCreated` or the `ComponentInstaller`.

When we make a new type of object, it needs to be enabled in a different way https://github.com/mapbox/mapbox-navigation-android/pull/6483/files#diff-1e15f303831c62824e88d37a6920faf7248d2396de4f5763f8524b15c855654bR146

Reply:
👍🏻